app/models/relation.rb

```diff
@@ -248,6 +248,9 @@ def preconditions_ok?(good_members = [])
       model = Kernel.const_get(m[0].capitalize)
       # get the element with that ID
       element = model.find_by(:id => m[1])
+      # if found, lock the element to ensure it can't be deleted until
+      # after the current transaction commits.
+      element.lock!("for share") unless element.nil?
```

Comment:
Should you fold that into the previous line as `model.lock("for share").find_by(:id => m[1])`? Otherwise there is a race between finding it and locking it where a deletion could still happen?